pymeasure/instruments/rohdeschwarz/fsl.py

```diff
@@ -302,15 +302,31 @@ def select_channel(self, channel_name):
         """
         self.write(f"INST:SEL '{channel_name}'")
 
-    # active_channel = Instrument.control("INST?", "INST %s", "To get the active channel")
+    @property
+    def active_channel(self):
+        return self.values("INST?")[0]
+
+    @active_channel.setter
+    def activate_channel(self, channel):
+        availabel_channels = [chan for chan in self.available_channels.keys()]
+        channel = strict_discrete_set(channel, availabel_channels)
+        self.write(f"INST '{channel}'")
+
+    def view_mode(self, view="SPL"):
+        """To get a MultiView of the open channels or return to the last active channel
+        Args:
+            view (str, optional): To view MultiView. Defaults to "SPL". To go back to the last
+            active channel use "SING".
+        """
+        self.write(f"DISP:FORM {view}")
 
     def rename_channel(self, current_name, new_name):
         """Renames an open channel
         Args:
             current_name (string): channel to be renamed
             new_name (String): new name of the channel
         """
-        channels = self.show_channels
+        channels = self.available_channels
         strict_discrete_set(current_name, list(channels.keys()))
```

Comment:
`self.available` channel is only called once so no need to assign a local variable, as stated above. `strict_discrete_set` returns the valid `current_name` so I think it should be overwritten here, like current_name = `strict_discrete_set(current_name, list(channels.keys()))`

It will throw an error in any case but I think it is cleaner that way.

Reply:
Resolved via [697127a](https://github.com/bleykauf/pymeasure/pull/4/commits/697127a14c501fae622095b3e2ec04cabfc76fdd)